kiali/kiali
# Use auth for check urls in Grafana & tracing

### Describe the change

- Use auth if defined for url version check for Tracing & Grafana, as the same way as Prometheus.

### Steps to test the PR

- Use an authenticated url for Grafana & tracing. 
- Use incorrect credentials and validate that kiali reports the service as unhealthy. 
E.x.
```grafana:
    url: https://grafana.url
    auth:
      insecure_skip_verify: true
      password: xxx
      type: basic
      use_kiali_token: false
      username: wrongUser
```
Grafana version is unknown. The error in the logs: 
`INF grafana version check failed: url=[https://grafana.url], code=[401]`


![image](https://github.com/kiali/kiali/assets/49480155/fcc8818b-fd3b-4423-b861-6084100af79e)

- Use correct credentials and validate that Kiali reports as healthy services. 
E.x.
```grafana:
    url: https://grafana.url
    auth:
      insecure_skip_verify: true
      password: xxx
      type: basic
      use_kiali_token: false
      username: correctUser
```
Grafana version is set:
![image](https://github.com/kiali/kiali/assets/49480155/ac62698f-2b5d-4320-bd74-d28ea70a1b4e)


### Automation testing

If applicable, explain the case scencarios covered by **unit / integration / e2e** tests created for this PR.

### Issue reference

Fixes https://github.com/kiali/kiali/issues/7475
